cupy/cuda/nccl.pyx

```diff
@@ -201,3 +212,13 @@ cdef class NcclCommunicator:
                                     count, <ncclDataType_t>datatype,
                                     self._comm, <driver.Stream>stream)
         check_status(status)
+
+    def check_async_error(self):
+        if NCCL_VERSION_CODE < 2400:
+            raise RuntimeError("ncclCommGetAsyncError is not available"
+                               " in this version")
+        cdef ncclResult_t asyncError = ncclSuccess
+        with nogil:
+            result = ncclCommGetAsyncError(self._comm, &asyncError)
```

Comment:
A tiny point: Is this `with nogil` required? 
In my understanding releasing and re-locking GIL is costly,  so it should be generally avoided in short-time functions.
My knowledge on cython is limited, so I might be wrong.

Reply:
The method's example use case is [this](https://github.com/NVIDIA/nccl/pull/140#issuecomment-429121534) and thus I assume this won't be a hot code path where locking cost matters. This method will also be called in error handling logic to get errors and the program is fated to abort sooner or later, (or the NCCL ring reconstructed). I thought this function might block, and cause deadlock in the whole process, but so far in NCCL 2.4 code doesn't seem to block. Not sure future versions of this function may or may not block, to get some shared state variable so I think it'd be safer to assume this function may block.

On the other hand `ncclCommAbort` and `ncclCommDestroy` does not release the GIL when they're called, but they're really the finalization functions  and everything involving NCCL must allow being blocked until finalization finishes in this case. Actually destroy blocks while abort does not, so the destroy method may have to release the GIL...

Reply:
OK, thank you for clarification!
Then how about leaving a brief comment on the declaration of `ncclCommGetAsyncError` about the intent of `nogil`?
Without knowing the intent, one would be tempted to remove it. Or perhaps someone could prove that it actually never blocks and remove it.